File: spark/cart-backend/src/main/java/com/mercadolibre/cart_backend/cart/tasks/GetCartTask.java
package com.mercadolibre.cart_backend.cart.tasks;

import com.mercadolibre.cart_backend.cart.entities.Cart;
import com.mercadolibre.cart_backend.cart.entities.CartItem;
import com.mercadolibre.cart_backend.core.TaskBase;

import java.util.Map;

/**
 * Created by fvitali on 8/8/16.
 */
public class GetCartTask extends TaskBase<Cart> {
	@Override
	public Cart execute(Map<String, Object> params) {
		super.execute(params);

		String cartId = (String) params.get("cartId");

		if (cartId.equals("667")) {
			return null;
		}

		Cart cart = new Cart(cartId);
		cart.getItems().add(new CartItem("1", 1, "My item 1", 13.28));
		cart.getItems().add(new CartItem("2", 2, "My item 2", 1024.36));
		cart.getItems().add(new CartItem("3", 1, "My item 3", 666));

		return cart;
	}
}

File: node.typescript/cart-backend/src/cart/tasks/getCartTask/default/native/v02.00.00/getCartTask.ts
import Task from '../../../../../../core/contracts/Task';
import Cart from '../../../../../entities/cart';
import CartItem from '../../../../../entities/cartItem';
import CartConfig_01_00_00 from '../../../../../config/CartConfig_01_00_00';
import { Promise } from 'es6-promise';
import CartService from '../../../../../services/CartService';
import UserService from '../../../../../services/UserService';
import ShippingService from '../../../../../services/ShippingService';
import LocalizationProvider from '../../../../../../framework/i18n/LocalizationProvider';
import GetCartTask_v01_00_00 from '../../v01.00.00/getCartTask'

export default class GetCartTask extends GetCartTask_v01_00_00 {

    public async execute(params: Map<string, any>): Promise<Cart> {
        return super.execute(params).then((cart: Cart) => {
            cart.price *= 2;
            return cart;
        })
    }
}